tabular/src/autogluon/tabular/trainer/abstract_trainer.py

```diff
@@ -37,13 +37,15 @@ class AbstractTrainer:
 
     def __init__(self, path: str, problem_type: str, eval_metric=None,
                  num_classes=None, low_memory=False, feature_metadata=None, k_fold=0, n_repeats=1,
-                 save_data=False, random_state=0, verbosity=2):
+                 weight_column=None, weight_evaluation=False, save_data=False, random_state=0, verbosity=2):
         self.path = path
         self.problem_type = problem_type
         self.feature_metadata = feature_metadata
         self.save_data = save_data
         self.random_state = random_state  # Integer value added to the stack level to get the random_state for kfold splits or the train/val split if bagging is disabled
         self.verbosity = verbosity
+        self.weight_column = weight_column
```

Comment:
I don't think it is necessary for trainer to know about what the weight column name is, same way it doesn't know about label column name.

Reply:
Will keep current internal implementation with weights bundled inside X for now as discussed. Future contributors may otherwise not remember to handle sample-weights when they are adding new functionality, so alternative may need careful design.